src/main/java/com/google/finapp/SpannerDaoImpl.java

```diff
@@ -55,6 +60,9 @@ public void createCustomer(ByteArray customerId, String name, String address)
   public void createAccount(
       ByteArray accountId, AccountType accountType, AccountStatus accountStatus, BigDecimal balance)
       throws SpannerDaoException {
+    if (balance.signum() == -1) {
+      throw new IllegalArgumentException("Account balance cannot be negative");
+    }
```

Comment:
Any of these `if` `throw` constructions throughout the PR can also be assertions. I chose this method so I could specifically use `IllegalArgumentException` to indicate that something the user input was incorrect, but if assertions are the preferred method, this can be switched for any or all of these instances.

Reply:
`IllegalArgumentException` looks right to me.